src/bundle.rs

```diff
@@ -18,267 +18,505 @@ use crate::{
-                synapse.to_string(),
-                to.to_label(),
+            let pre_gid = pop_to_gid[pre] as i64;
+            let post_gid = pop_to_gid[post] as i64;
+            for Connection {
+                from,
+                to,
                 weight,
                 delay,
-            ));
+            } in connections
+            {
+                let pre_cell_id = &net.populations[pre].component;
+                let threshold = cell_to_threshold[pre_cell_id];
+                let from_gid = pre_gid + from.cell;
+                let to_gid = post_gid + to.cell;
+                gid_to_detectors
+                    .entry(from_gid)
+                    .or_default()
+                    .push((from.segment, from.fraction.clone(), threshold));
+                gid_to_synapses.entry(to_gid).or_default().push((
+                    to.segment,
+                    to.fraction.clone(),
+                    synapse.clone(),
+                ));
+                gid_to_connections
+                    .entry(to_gid)
+                    .or_insert_with(Vec::new)
+                    .push((
+                        from_gid,
+                        from.to_label(),
+                        synapse.to_string(),
+                        to.to_label(),
+                        *weight,
+                        *delay,
+                    ));
+            }
         }
+
+        let mut i_clamps = Map::new();
+        let mut poisson_generators = Map::new();
+        let regular_generators = Map::new();
+        for (lbl, stimulus) in stimuli {
+            match stimulus {
+                Input::Pulse(delay, dt, stop) => {
+                    i_clamps.insert(lbl.to_string(), (*delay, *dt, *stop));
+                }
+                Input::Poisson(syn, avg, wgt) => {
+                    poisson_generators.insert(lbl.to_string(), (syn.to_string(), *avg, *wgt));
+                }
+            }
+        }
+
+        Ok(SimulationData {
+            gid_to_cell,
+            cell_to_morph: cell_to_morph.clone(),
+            gid_to_inputs,
+            gid_to_synapses,
+            gid_to_detectors,
+            gid_to_connections,
+            i_clamps,
+            regular_generators,
+            poisson_generators,
+            count,
+        })
     }
+}
+
+fn mk_bash() -> Result<String> {
+   Ok("#!/usr/bin/env bash
+
+set -euxo pipefail
+
+cmake -S . -B build
+cmake --build build
+cp build/main .
+arbor-build-catalogue local cat
+./main $*
+".to_string())
+
+}
+
+fn mk_cmake() -> Result<String> {
+    Ok("
+cmake_minimum_required(VERSION 3.22)
+project(arbor-ring LANGUAGES CXX)
+set(CMAKE_CXX_STANDARD 17)
+
+find_package(arbor REQUIRED)
+find_package(nlohmann_json 3.11.0 REQUIRED)
+
+add_executable(main main.cxx)
+target_link_libraries(main PUBLIC arbor::arbor arbor::arborio arbor::arborenv nlohmann_json::nlohmann_json)
+".to_string())
+}
+
+fn mk_main_cxx() -> Result<String> {
+    Ok("
```

Comment:
With the templates becoming a string, we really should move these to external files and include with `include_str!("simulationmain.cpp")` imo

Reply:
That's an excellent idea!